### PR TITLE
Fix dedup logic - filter out vulnerabilities from nvd namespaces if t…

### LINKED ISSUE #893: same non-os vulnerability reported in github and nvdv2 namespaces
A recent change to the dedup logic is causing the same CVE to be listed both in nvdv2 and githubu namespace. This is a regression from 0.9.0 

```
$ anchore-cli image vuln docker.io/node:latest non-os
Vulnerability ID           Package              Severity        Fix          CVE Refs                Vulnerability URL                                                Type          Feed Group                    Package Path
CVE-2017-9462              mercurial-4.0        High            None         CVE-2017-9462           https://nvd.nist.gov/vuln/detail/CVE-2017-9462                   python        nvdv2:cves                    /usr/lib/python2.7/dist-packages/mercurial
GHSA-ghjx-3jg5-h6r2        mercurial-4.0        High            4.1.3        CVE-2017-9462           https://github.com/advisories/GHSA-ghjx-3jg5-h6r2                python        github:python                 /usr/lib/python2.7/dist-packages/mercurial
```

### PR DESCRIPTION
**Pull request by nightfurys**
…hey are reported by other namespaces

Fixes #893

Signed-off-by: Swathi Gangisetty <swathi@anchore.com>

### CODE REVIEW

**Review by dakaneye — APPROVED** [review]
looks right to me! We need unit tests for this logic but can be added later

**Review by zhill — COMMENTED** [review]
I'd like to see some tests for this code, the logic here is getting complex. Can you add a couple of tests of the outer function to show the expected behavior.

File: anchore_engine/services/apiext/api/controllers/utils.py [hunk 1]
```diff
@@ -225,3 +229,245 @@ def validate_image_add_source(analysis_request_dict, api_schema):
...
+
+
+def make_response_vulnerability(vulnerability_type, vulnerability_data):
+    ret = []
+
+    if not vulnerability_data:
+        logger.warn("empty query data given to format - returning empty result")
+        return ret
+
+    eltemplate = {
```
Comment (dspalmer99) [inline anchore_engine/services/apiext/api/controllers/utils.py:310]: What does "el" stand for throughout here?

Comment (zhill) [reply anchore_engine/services/apiext/api/controllers/utils.py:310]: this is copy-pasted code, el = element

**Review by zhill — APPROVED** [review]
looks good. thanks for adding tests much clearer now.

**Pull request merged**